frontend/www/js/omegaup/components/contest/AddProblem.vue

```diff
@@ -195,15 +205,45 @@ export default class AddProblem extends Vue {
   publishedRevision: null | types.ProblemVersion = null;
   selectedRevision: null | types.ProblemVersion = null;
 
-  onSubmit(): void {
-    if (this.useLatestVersion) {
-      this.$emit('get-versions', {
-        target: this,
-        request: { problemAlias: this.alias },
-      });
-    } else {
-      this.onAddProblem();
+  setInitialRevisionValues(): void {
+    this.versionLog = [];
+    this.selectedRevision = this.publishedRevision = null;
+    this.useLatestVersion = true;
+  }
+
+  get problemsMapping(): MappedProblems {
+    let problemsMapping: MappedProblems = {};
+    for (const problem of this.problems) {
+      const publishedCommitHash = problem.commit;
+      problemsMapping[problem.alias] = {
+        versions: problem.versions.log,
+        mappedVersions: {},
+        published: problem.commit,
+        useLastVersion: false,
+      };
+      for (const [index, version] of problem.versions.log.entries()) {
+        problemsMapping[problem.alias].mappedVersions[version.commit] = version;
+        if (index === 0 && version.commit === publishedCommitHash) {
+          problemsMapping[problem.alias].useLastVersion = true;
+        }
+      }
     }
+    return problemsMapping;
+  }
+
+  onGetVersions(problemAlias: string): void {
+    this.versionLog = this.problemsMapping[problemAlias].versions;
+    const published = this.problemsMapping[problemAlias].published;
+    const revision = this.problemsMapping[problemAlias].mappedVersions[
+      published
+    ];
+    this.selectedRevision = this.publishedRevision = revision;
+    this.useLatestVersion = this.problemsMapping[problemAlias].useLastVersion;
```

Comment:
creo que es mejor dejar `useLatestVersion` siempre deshabilitado al momento de actualizar problemas, y pedirle al usuario que siempre elija explícitamente la versión (o explícitamente elija que quiere usar la última).

```suggestion
    this.useLatestVersion = false;
```

Reply:
Listo.